src/main/scala/org/broadinstitute/hail/Utils.scala

```diff
@@ -327,6 +327,49 @@ class RichSparkContext(val sc: SparkContext) extends AnyVal {
       .map(l => Line(l, None, file))
 }
 
+class HeadRDD[T](prev: RDD[T], num: Int, exact: Boolean)(implicit tct: ClassTag[T]) extends RDD[T](prev) {
+
+  private var nLastPart = 0
+  private var lastPart = 0
+
+  def getPartitions: Array[Partition] = {
+    val buf = new mutable.ArrayBuffer[Partition]
+
+    var nvp = prev.sparkContext.runJob(prev,(it: Iterator[T]) => it.size, Seq(0))
+    if(exact) {
+      while (nvp.sum < num && nvp.size < prev.partitions.length) {
+        nvp = nvp ++ prev.sparkContext.runJob(prev, (it: Iterator[T]) => it.size,
```

Comment:
This `++` makes it quadratic in the number of times you have to call `runJob`.  While we hope that is small, quadratic makes me nervous.  Although I said delete `buf`, doing `buf ++= ...` will make it linear.